lib_xua/src/core/clocking/clockgen.xc

```diff
@@ -557,6 +566,21 @@ void clockGen ( streaming chanend ?c_spdif_rx,
                     /* Returns 1 if valid clock found */
                     valid = validSamples(adatCounters, CLOCK_ADAT);
 #if XUA_USE_SW_PLL
+                    if(clkMode == CLOCK_ADAT)
+                    {
+                        unsigned expected_mclk = \
+                            ((clockFreq[CLOCK_ADAT] == 48000) || (clockFreq[CLOCK_ADAT] == 96000) || (clockFreq[CLOCK_ADAT] == 192000)) ? MCLK_48 : MCLK_441;
```

Comment:
is this not just

selected_mclk = (clockFreq[CLOCK_ADAT] % 44100) ? MCLK_48: MCLK_441; 

or removing the extra var and just using the SR:

if (selected_sample_rate % 44100 == 0 && clockFreq % 44100 == 0) 
|| (selected_sample_rate % 48000 == 0 && clockFreq % 48000 == 0) ?

Reply:
hmm, I thought the mod operator would not be optimal in terms of cycles so decide to do it this way

Reply:
Okay, speed shouldn't really matter here. Was thinking more about readability. But it's okay I think.